plugins/link/dialogs/anchor.js

```diff
@@ -30,12 +30,38 @@ CKEDITOR.dialog.add( 'anchor', function( editor ) {
 			element = element.getParent();
 		}
 
+		// If anchor is exist and has some styles go to the closest parent <a> tag. (#3863)
+		if ( element && !element.is( 'a' ) ) {
+			element = element.getAscendant( 'a' ) || element;
+		}
+
 		if ( element && element.type === CKEDITOR.NODE_ELEMENT &&
 			( element.data( 'cke-real-element-type' ) === 'anchor' || element.is( 'a' ) ) ) {
 			return element;
 		}
 	}
 
+	// Checks if range contains some nested anchors and remove it.
+	function checkNestedAnchors( range ) {
+		var newRange = range.clone();
+		newRange.enlarge( CKEDITOR.ENLARGE_ELEMENT );
```

Comment:
You can make it a one-liner `...clone().enlarge( ... );` and then merge with below vars block.

Reply:
TBH I don't see a way how this could be inlined. `enlarge` returns `undefined`, so inlining it makes `newRange` undefined. I'm marking this conversation as resolved.